cprover_bindings/src/irep/irep.rs

```diff
@@ -44,6 +44,10 @@ impl Irep {
         }
     }
 
+    pub fn with_comment<T: Into<InternedString>>(self, c: T) -> Self {
```

Comment:
cool. doc comment? I know I'm the one who vaguely waved you in the direction of "putting a comment on an irep" but did you dig into exactly how this works? It'd be good to explain I think...

Reply:
That one is a bit blurry actually. It will add a comment directly to the irep.
As of now, comments are usually in the "location" sub, and here I'm making the decision to put it outside.
Not sure that's the best option, but the location is handled by Expr::to_irep not ExprValue::to_irep, so I can't propagate it up without a big refactoring that would clutter the entire thing for one use case.

I do think we need a more standard way of writing comments like that for alternative back-ends